src/internal/Modal.spec.js

```diff
@@ -51,6 +51,103 @@ describe('<Modal />', () => {
       assert.strictEqual(modal.is('div'), true, 'should be a div');
       assert.strictEqual(modal.hasClass(classes.modal), true, 'should have the modal class');
     });
+
+    describe('handlers', () => {
+      let instance;
+
+      before(() => {
+        instance = wrapper.instance();
+      });
+
+      describe('focus', () => {
+        before(() => {
+          instance.modal = spy();
+          instance.modal.lastChild = spy();
+          instance.modal.lastChild.setAttribute = spy();
+          instance.modal.lastChild.focus = spy();
+        });
+
+        describe('modalContent has tabIndex attribute', () => {
+          let consoleError;
+          let consoleErrorSpy;
+
+          before(() => {
+            instance.modal.lastChild.hasAttribute = stub().returns(true);
+            consoleError = console.error; // eslint-disable-line no-console
+            consoleErrorSpy = spy();
+            // $FlowFixMe
+            console.error = consoleErrorSpy; // eslint-disable-line no-console
+            instance.focus();
+          });
+
+          after(() => {
+            instance.modal.lastChild.hasAttribute.reset();
+            instance.modal.lastChild.focus.reset();
+            consoleErrorSpy.reset();
+            // $FlowFixMe
+            console.error = consoleError; // eslint-disable-line no-console
+          });
+
+          it('should call hasAttribute with tabIndex', () => {
+            assert.strictEqual(instance.modal.lastChild.hasAttribute.callCount, 1);
+            assert.strictEqual(instance.modal.lastChild.hasAttribute.calledWith('tabIndex'), true);
+          });
+
+          it('should not call setAttribute', () => {
+            assert.strictEqual(instance.modal.lastChild.setAttribute.callCount, 0);
+          });
+
+          it('should not call console.error', () => {
+            assert.strictEqual(consoleErrorSpy.callCount, 0);
+          });
+
+          it('should call focus', () => {
+            assert.strictEqual(instance.modal.lastChild.focus.callCount, 1);
+          });
+        });
+
+
+        describe('modalContent does not have tabIndex attribute', () => {
+          let consoleError;
+          let consoleErrorSpy;
+
+          before(() => {
+            instance.modal.lastChild.hasAttribute = stub().returns(false);
+            consoleError = console.error; // eslint-disable-line no-console
+            consoleErrorSpy = spy();
+            // $FlowFixMe
```

Comment:
I think that having a factory for overriding the `console.error` could be useful.
That's something we gonna need to repeat if we want to increase the coverage.